README.md

```diff
@@ -1,131 +1,13 @@
-* [API:RecentChanges](https://www.mediawiki.org/wiki/API:RecentChanges)
-  * [get_recent_changes.py](python/get_recent_changes.py): get the three most recent changes with sizes and flags
-* [API:Querypage](https://www.mediawiki.org/wiki/API:Querypage)
-  * [get_querypage_list.py](python/get_querypage_list.py): List first 10 pages which are uncategorized
-* [API:SetPageLanguage](https://www.mediawiki.org/wiki/API:SetPageLanguage)
-  *  [set_page_language.py](python/set_page_language.py): change page language
-* [API:Embeddedin](https://www.mediawiki.org/wiki/API:Embeddedin)
-  * [get_embedded_pages.py](python/get_embedded_pages.py): get all page(s) that embed a page
-* [API:Rollback](https://www.mediawiki.org/wiki/API:Rollback)
-  * [rollback.py](python/rollback.py): rollback the last edits made to a given page
-* [API:Allfileusages](https://www.mediawiki.org/wiki/API:Allfileusages)
-  * [get_allfileusages.py](python/get_allfileusages.py): list of all file usages
-* [API:Protect](https://www.mediawiki.org/wiki/API:Protect)
-  * [protect.py](python/protect.py): Change the protection level of a given page
-* [API:Upload](https://www.mediawiki.org/wiki/API:Upload)
-  * [upload_file_directly.py](python/upload_file_directly.py): upload a file directly from the system
-  * [upload_file_from_url.py](python/upload_file_from_url.py): upload a file from a URL 
-  * [upload_file_in_chunks.py](python/upload_file_in_chunks.py): upload a file in chunks
-* [API:Pagepropnames](https://www.mediawiki.org/wiki/API:Pagepropnames)
-  * [get_pagepropnames.py](python/get_pagepropnames.py): List all page property names in use on the wiki
+# MediaWiki Action API Code Samples
+The MediaWiki action API is a web service that allows access to some wiki-features like authentication, page operations, and search. It can provide Meta information about the wiki and the logged-in user.
 
-### Search 
-* [API:Search](https://www.mediawiki.org/wiki/API:Search)
-  * [search.py](python/search.py): search for a title or a text
-* [API:Geosearch](https://www.mediawiki.org/wiki/API:Geosearch)
-  * [geosearch.py](python/geosearch.py): search for pages nearby
-  * [geoimagesearch.py](python/geoimagesearch.py): search for pages nearby with images
-  * [geocoordinates.py](python/geocoordinates.py): obtain coordinates for wiki pages nearby
-* [API:Opensearch](https://www.mediawiki.org/wiki/API:Opensearch)
-  * [opensearch.py](python/opensearch.py): fetch results in an opensearch format
-* [API:Prefixsearch](https://www.mediawiki.org/wiki/API:Prefixsearch)
-  * [prefixsearch.py](python/prefixsearch.py): perform a prefix search for page titles
-* [API:Languagesearch](https://www.mediawiki.org/wiki/API:Languagesearch)
-  * [languagesearch.py](python/languagesearch.py): search for a language 
+The code snippets in several programming languages are developed to assist developers for easy use of MediaWiki action API's.
```

Comment:
Rephrase to "Code snippets in several programming languages are developed to assist developers for easy use of various modules of the MediaWiki Action API."